components/vfs/src/android_aasset_manager.cpp

```diff
@@ -145,34 +116,31 @@ StackErrorPtr AndroidAAssetManager::read_chunk(const std::string &file_path, con
 
 	if (!asset)
 	{
-		return StackError::unique("failed to find file: " + file_path, "vfs/android_aasset_manager.cpp", __LINE__);
+		throw std::runtime_error("vfs/android_aasset_manager.cpp line" + std::to_string(__LINE__) + "failed to find file: " + file_path);
 	}
 
 	size_t size = AAsset_getLength(asset);
 
 	if (offset + count > size)
 	{
-		return StackError::unique("requested chunk out of range", "vfs/android_aasset_manager.cpp", __LINE__);
+		throw std::runtime_error("vfs/android_aasset_manager.cpp line" + std::to_string(__LINE__) + "requested chunk out of range");
 	}
 
-	auto a_blob = std::make_shared<StdBlob>();
-	a_blob->buffer.resize(size, 0);
+	std::vector<uint8_t> blob(size, 0);
 
 	AAsset_seek(asset, offset, 0);
-	AAsset_read(asset, const_cast<void *>(reinterpret_cast<const void *>(a_blob->buffer.data())), size);
+	AAsset_read(asset, const_cast<void *>(reinterpret_cast<const void *>(blob.data())), size);
 
 	AAsset_close(asset);
 
-	*blob = a_blob;
-
-	return nullptr;
+	return blob;
 }
 
-size_t AndroidAAssetManager::file_size(const std::string &file_path)
+size_t AndroidAAssetManager::file_size(const std::string &file_path) const
 {
 	if (!asset_manager)
 	{
-		return 0;
+		throw std::runtime_error("vfs/android_aasset_manager.cpp line" + std::to_string(__LINE__) + "AAsset Manager not initialized");
```

Comment:
Does it make sense to make this a macro and use something like:
#define ERROR_STR(...) (( __FILE__ ":" __LINE__ " "  __VA_ARGS__))

Something to make the repeated tasks like listing out the file name a little less required and only capture the actual error response is what I'm looking for.  Can change the above as needed.

Reply:
This should be addressed in a new PR, I will create an issue to follow up on this.

Reply:
I'm not a fan of having the filename hard-wired like this. When someone wants to change the filename at some point in the future, it's going to be a little irritating to have to change all these too.

__FILE__ seems like a perfectly reasonable thing to use here, and hiding that inside a macro seems fine to me. If you really don't want the full path to appear in the output then maybe do some processing on the path to strip the front off.